### PR TITLE
Improves Exception Handling in the Faban Client

### LINKED ISSUE #409: Illegal run status (unknown) from Faban
during execution of an experiment following error occurred and stopped the progress

```
benchflow-experiment-manager_1  | java.util.concurrent.ExecutionException: cloud.benchflow.faban.client.exceptions.IllegalRunStatusException: RunId jespertest_TestExploreUsersOnCamunda750_5_3.1Ureturned illegal run status UNKNOWN
benchflow-experiment-manager_1  | 
benchflow-experiment-manager_1  | 	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
benchflow-experiment-manager_1  | 	at java.util.concurrent.FutureTask.get(FutureTask.java:192)
benchflow-experiment-manager_1  | 	at cloud.benchflow.experimentmanager.scheduler.ExperimentTaskScheduler.handleDetermineAndExecuteTrials(ExperimentTaskScheduler.java:193)
benchflow-experiment-manager_1  | 	at cloud.benchflow.experimentmanager.scheduler.ExperimentTaskScheduler.handleRunningState(ExperimentTaskScheduler.java:156)
benchflow-experiment-manager_1  | 	at cloud.benchflow.experimentmanager.scheduler.ExperimentTaskScheduler.handleExperimentState(ExperimentTaskScheduler.java:95)
benchflow-experiment-manager_1  | 	at cloud.benchflow.experimentmanager.scheduler.ExperimentDispatcher.run(ExperimentDispatcher.java:69)
benchflow-experiment-manager_1  | 	at java.lang.Thread.run(Thread.java:748)
benchflow-experiment-manager_1  | Caused by: cloud.benchflow.faban.client.exceptions.IllegalRunStatusException: RunId jespertest_TestExploreUsersOnCamunda750_5_3.1Ureturned illegal run status UNKNOWN
```

We should handle the `UNKNOWN` case without any runtime exception in [RunStatus.java](https://github.com/benchflow/benchflow/blob/devel/benchflow-faban-client/src/main/java/cloud/benchflow/faban/client/responses/RunStatus.java)

### PR DESCRIPTION
**Pull request by VincenzoFerme**
Fixes #409

### CODE REVIEW

File: benchflow-faban-client/src/main/java/cloud/benchflow/faban/client/responses/RunStatus.java [hunk 1]
```diff
@@ -42,9 +45,13 @@ public RunStatus(String statusCode, RunId runId) {
       case "DENIED":
         this.status = StatusCode.DENIED;
         break;
+      case "UNKNOWN":
+        this.status = StatusCode.UNKNOWN;
+        break;
       default:
         throw new IllegalRunStatusException(
-            "RunId " + runId + "returned illegal run status " + statusCode);
+            "RunId " + runId + "returned illegal run status " + formattedStatusCode,
```
Comment (jeppe-style) [inline benchflow-faban-client/src/main/java/cloud/benchflow/faban/client/responses/RunStatus.java:53]: I would not throw an exception here but rather just set the `status` to `UNKNOWN`

**Review by jeppe-style — COMMENTED** [review]
It was a bit difficult to review since many changes were done in the previous PR.
General comment: in which cases do we want a `RuntTimeException` and not a checked exception? I have earlier added something about exceptions in the temporary documentation: https://github.com/VincenzoFerme/benchflow-temporary-documentation/blob/master/development/exceptions.md

File: benchflow-faban-client/src/main/java/cloud/benchflow/faban/client/responses/RunStatus.java (hunk 1 above)
Comment (VincenzoFerme) [reply benchflow-faban-client/src/main/java/cloud/benchflow/faban/client/responses/RunStatus.java:53]: The point is that `UNKNOWN` is a possible status of the client.

**Comment by VincenzoFerme** [comment]
So basically now what we have is:

- `RuntimeException` when there is no possibility for the client to recover it. For example in case of unknown error in interacting with Faban, or in case of `BenchmarkNameNotFound`, `IllegalRunId`, `MalformedURI`. 
- `CheckedException (Throwable)` when the client can recover it. For example: in case of an [unexpected `RunStatus`](https://github.com/benchflow/benchflow/pull/442/files#diff-4edade294df1b463ebb31a6487a9fe4bR52), in which case we also return to the caller, the formatted run status we got from Faban that leads to the exception.

**Comment by VincenzoFerme** [comment]
We agreed that we should not have `RuntimeException` because the client has always to be able to decide how to react to some exception, unless the Faban client can handle the exception.

**Comment by VincenzoFerme** [comment]
@jeppe-style I updated the pull with what we discussed. It should be ready for a final review.

File: benchflow-faban-client/src/main/java/cloud/benchflow/faban/client/exceptions/FabanClientThrowable.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package cloud.benchflow.faban.client.exceptions;
+
+/**
+ * @author vincenzoferme.
+ */
+public class FabanClientThrowable extends Throwable {
```
Comment (jeppe-style) [inline benchflow-faban-client/src/main/java/cloud/benchflow/faban/client/exceptions/FabanClientThrowable.java]: should this not be a subclass of [`Exception`](https://docs.oracle.com/javase/7/docs/api/java/lang/Exception.html)?

Comment (VincenzoFerme) [reply benchflow-faban-client/src/main/java/cloud/benchflow/faban/client/exceptions/FabanClientThrowable.java]: Right, fixed in https://github.com/benchflow/benchflow/pull/442/commits/6658c1b28264aa730db3b8804c2187df8254f0c0

**Review by jeppe-style — DISMISSED** [review]

**Comment by VincenzoFerme** [comment]
@jeppe-style I interactively rebased this pull request. Can you approve it so I can merge?

**Review by jeppe-style — APPROVED** [review]

**Pull request merged**